### ISSUE TITLE
模型问题

### ISSUE THREAD

**Issue by xinlong-image** [LeeWise9/Segmentation-Based-Surface-Defect-Detection#2]
你好，有两个问题请教：

1. 原文中segmentation和classification是分开训练的，你这里同时训练，是否会影响效果？
2. 在build_model函数中，seg_output是softmax归一化后的像素概率数组，seg_conv4是归一化之前的logits数组， 它们放在一起操作是否合适？

谢谢！

**Comment by LeeWise9** [comment]
1.原文说分开训练是因为网络难以训练。我更改了输入尺寸，计算压力变小了。不分开训练同样可以把网络训练得很好。
2.没有对比同时训练和分开训练的效果差异，因为没必要。
3.你的第二个问题，我猜你是想问 dec_conv1 = concatenate([seg_conv4,seg_output],axis=3)  这一步直接把两者拼接起来是否合适。
其实原文中提出把这两者连接在一起就很奇怪，就说 channel 的比例（原文1024:1）就注定mask的权重很小。如果你觉得不妥，可以在 mask 后添加卷积层扩充 channel。
4.另，效果更好的语义分割模型还有 FCN、ENet、DeepLab、PSPNet、ERFNet等。

**Comment by xinlong-image** [comment]
感谢你的详细回复！
1. 我感觉分开训练有一个目的是尽量不让segmentation受classification的影响，因为这一步结果对分类正确比较重要。
3. 我的意思是seg_output是两个卷积核产生，并且由softmax进行了归一化处理，直接和未做归一化的tensor粘合在一起是否合理。（原文粘合以后第二维张量是1025个，你这里是1026个）。我试着用1个卷积核sigmoid处理，得到的效果比较好。当然这样做也需要把segment model的output格式做修改。
4. 非常感谢你推荐的其它模型，有时间学习一下！

**Comment by LeeWise9** [comment]
1.对，我的经验是网络整体训好了把 classification 再单独加训，提升分类准确率
2.合理与否我没研究过，你可以做个对比实验。seg_output 用了两个卷积核是为了适配更多种类的缺陷数据。

**Comment by JH-Lam** [comment]
有几个问题请教一下：
1.觉得多了个softmax操作来串接问题不大，你就认为是NN多/少学习了这个操作
2.确认如 @xinlong-image 所说比原文多了一个 channels
3.据了解 这个paper比 FCN已经是比较新的了，不清楚是否真的效果更好
4.其实我认为这个decision network 是否有必要？我看有些人是不输出这个结果的。一方面可能存在不一致：如pred cls是1，但未必mask是存在瑕疵吧？
5.一般来说不会对label mask操作resize操作，因为它涉及插值问题，这样可能会填入不恰当的值。但我看到在data_manager.py中你是同时应用了src,label的

盼回复，thanks
@LeeWise9

**Comment by chujxchu** [comment]
您好，想问下如何看训练结果，运行test只有Namespace(cuda=True, img_height=704, img_width=256, test_dec_epoch=60, test_seg_epoch=60)
